api/src/main/java/org/apache/apex/api/ApexPlugin.java

```diff
@@ -18,12 +18,15 @@
  */
 package org.apache.apex.api;
 
+import org.apache.hadoop.classification.InterfaceStability;
+
 import com.datatorrent.api.Component;
 
 /**
  * Marker interface for ApexPlugins.
  * @param <T>
  */
+@InterfaceStability.Evolving
 public interface ApexPlugin<T extends ApexPluginContext> extends Component<T>
```

Comment:
Currently we have two set of plugins one runs at setup time to examine the DAG before it is launched and one to listen to various events at stram during runtime. Marking it as evolving as this is initial support and the interface might change in future if we need to add common functions, such as getVersion() , getName(). There were just two interfaces hence kept it in api package. do you suggest creating subpackage for plugins?